atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbTransactionStoreFactory.java

```diff
@@ -138,6 +140,8 @@ public static AtlasDbTransactionStoreFactory createFromConfig(
                 .globalMetricsRegistry(metricsManager.getRegistry())
                 .globalTaggedMetricRegistry(metricsManager.getTaggedRegistry())
                 .runtimeConfigSupplier(atlasDbRuntimeConfig)
+                .defaultTimelockClientFactory(
```

Comment:
This is 🤮 , but I lack to see better alternatives. One would want to pass in their own `LockAndTimestampServices`, but the wiring here is really ugly, and not trivial to do correctly (nor I couldn't find a case of ANYONE doing this). Furthermore, I tried multiple attempts at potentially exposing `LockAndTimestampServices` in the transaction manager, or it's factory, so that it could do the majority of the living for us, and that was also not trivial/possible. 

Thus, I've settled on the approach of providing us control over how to create the timelock client, which is something we want anyways.

Reply:
This seems fine to me - as you note it's unnatural, but I'm not sure exposing a client or LATS from the transaction manager is a good idea